modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/subscription/SubscriptionManagerImpl.kt

```diff
@@ -350,10 +359,10 @@ class SubscriptionManagerImpl @Inject constructor(
         subscriptionStatus.accept(Optional.empty())
     }
 
-    override fun isFreeTrialEligible() = freeTrialEligible
+    override fun isFreeTrialEligible(tier: Subscription.SubscriptionTier) = freeTrialEligible[tier] ?: false
 
-    override fun updateFreeTrialEligible(eligible: Boolean) {
-        freeTrialEligible = eligible
+    override fun updateFreeTrialEligible(tier: Subscription.SubscriptionTier, eligible: Boolean) {
```

Comment:
It looks like we're only ever calling this with an `eligible` value of `false`. Do we want it to always be false?

Reply:
Good catch! While the updated value should be always true when certain conditions are met, the initial value should have been true. Updated in 0d762d3.